app/org/sagebionetworks/bridge/models/schedules/Schedule.java

```diff
@@ -163,6 +165,14 @@ public boolean getPersistent() {
         }
         return false;
     }
+    @JsonIgnore
+    @DynamoDBIgnore
+    public TaskScheduler getScheduler() {
+        if (getCronTrigger() != null) {
+            return new CronTaskScheduler(this);
+        }
+        return new IntervalTaskScheduler(this);
+    }
```

Comment:
We don't want our objects to be structs. In this case, the choice of scheduler depends entirely on the data in scheduler, so it seems like a reasonable schedule responsibility. Later if I change schedule, it's easier to consider what impact this has on scheduler selection, and it's one less factory class.